### PR TITLE
Resources for init containers

### PR DESCRIPTION
**Pull request by marcinbojko**
Resources are being assigned only to regular pod. This makes op-scim-bridge impossible to deploy in any namespace with already assigned resources.
Adding scim.init section (for possible future changes for init-containers) fixes this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ The values are split into two sections:
 | nodeSelector | object | `{}` | Node selector for SCIM bridge pod. |
 | affinity | object | `{ "podAntiAffinity": {} }` | Affinity for SCIM bridge pod. By default we configure pod anti-affinity to ensure redis and SCIM bridge pods are not scheduled on the same node. |
 | tolerations | list | `[]` | Tolerations for SCIM bridge pod. |
+| init| object | `{}`| Resources for init containers|
```
Comment (devillexio) [inline README.md:76]:
```suggestion
| initResources | object | `{}` | Resources for init containers. |
```

File: op-scim-bridge/templates/deployment.yaml [hunk 2]
```diff
@@ -63,6 +63,10 @@ spec:
           volumeMounts:
             - mountPath: /home/opuser/.op
               name: {{ tpl .Values.scim.name . }}-credentials
+          {{ with .Values.scim.init.resources }}
```
Comment (devillexio) [inline op-scim-bridge/templates/deployment.yaml:66]:
```suggestion
          {{ with .Values.scim.initResources }}
```

File: op-scim-bridge/values.yaml [hunk 3]
```diff
@@ -88,7 +88,8 @@ scim:
           topologyKey: topology.kubernetes.io/zone
   # tolerations sets the tolerations for the SCIM bridge pod
   tolerations: []
-
+  init:
```
Comment (devillexio) [inline op-scim-bridge/values.yaml:91]:
```suggestion
  # initResources sets the resources for the init containers
```

File: op-scim-bridge/values.yaml [hunk 4]
```diff
@@ -88,7 +88,8 @@ scim:
...
+    resources: {}
```
Comment (devillexio) [inline op-scim-bridge/values.yaml:92]:
```suggestion
  initResources: {}
```

**Review by devillexio — COMMENTED** [review]
Hi @marcinbojko! 👋

Thanks so much for logging an issue and submitting a PR with a fix. The changes look good overall. I made some minor suggestions, but the overall idea is to collapse `init.resources` into `initResources`.

**Comment by marcinbojko** [comment]
> Hi @marcinbojko! wave
> 
> Thanks so much for logging an issue and submitting a PR with a fix. The changes look good overall. I made some minor suggestions, but the overall idea is to collapse `init.resources` into `initResources`.

I understand. But somewhere in future, you'd like to change for example init container Image.
Wouldn't be more convenient to NOT introduce variables based on yaml path but simply expanf existing init?

Let's call it initContainers instead of init.


Today you'll have:

```yaml
initContainers:
  resources: {}
```

in the future,wanting to add image variable:
```yaml
initContainers:
  ressources: {}
  image: {}
```
You won't have to introduce new variable: initImage

**Comment by devillexio** [comment]
Great point! I've updated the suggestions to use `initContainers` instead of `init`, but keep the nested `resources`.

Would you mind resolving the conflicts and reviewing the suggestions?

I'll do another review then and we can get this merged in. Thanks again for your effort. 🙌

**Comment by marcinbojko** [comment]
Sir, yes sir.

File: op-scim-bridge/templates/deployment.yaml (hunk 2 above)
Comment (devillexio) [reply op-scim-bridge/templates/deployment.yaml]: I think we unfortunately unintentionally removed the config from the `deployment.yaml`, would you mind adding it back?

I believe it should be:
```yaml
          {{ with .Values.scim.initContainers.resources }}
          resources:
            {{- toYaml . | nindent 12 }}
          {{ end }}
```

Comment (marcinbojko) [reply op-scim-bridge/templates/deployment.yaml]: I am sorry, it was my fault. I'll try to get it back, I am a little confused by the flow of commits to commits.

**Review by devillexio — APPROVED** [review]
Looks good to me! :tada: Thank you for submitting the PR. 🙌

File: op-scim-bridge/templates/deployment.yaml (hunk 2 above)
Comment (devillexio) [reply op-scim-bridge/templates/deployment.yaml]: No problem at all. Thanks for fixing it.

**Comment by devillexio** [comment]
Ah, I realized that the required `lint-and-test` check run via GitHub actions will only run when someone from our team submits the PR. I'm going to submit these exact changes in a PR and get it merged. I'll close this MR and related issue when it's done.

Thanks again for your effort!

**Pull request closed without merging**